### ISSUE TITLE
Shave Only Delimits by a Space.

### ISSUE THREAD

**Issue by CJames95** [yowainwright/shave#411]
## What Is the issue?

When shave interacts with a string of text, it delimits the text by spaces which means that if a different character such as a newline is the only character separating two words, it does not create two tokens. Instead it creates a single token with the newline character between the words. 

This means that it may shave a word from the end of a line if the word after the newline character would be shaved but the word before the newline character would otherwise fit.

## Provide issue context below using code examples, images, or links

`const example = 'I am a cow.\nHear me moo!'`

Currently shave would group `cow.\nHear` as a single token when it would make more sense to have it split the tokens as `cow.` and `Hear`. This is especially a problem for a textarea where new lines are possible and the user may not realize that their input would be shaved in a way that is likely not expected. 

**Edit:** To be more clear, the issue is when `I am a cow.` would be the final line before shave removed the rest of the string. The string currently is tokenized like this `['I', 'am', 'a', 'cow.\nHear', 'me' 'moo!']`. Since the final word on the line should be `cow.` it doesn't actually appear because `Hear` would be truncated therefore `cow.\nHear` is all truncated causing the sentence to appear as `I am a...` on the final line.

----

> Read about references issues [here](https://help.github.com/articles/closing-issues-using-keywords/). Provide paragraph text responses to each header.

**Comment by CJames95** [comment]
I added this functionality by turning your package into a helper file in my project. I went ahead and made a pull request on #412 that has my changes. I may have not copied all the changes since I had to do more for my project, but I'd be happy to revise or fix what I missed if I did.

**Comment by yowainwright** [comment]
Can you provide a code sandbox of some sort to the issue you have? 
I have some concerns with this feature request but want to ensure I'm understanding you fully!
~Thanks!

**Comment by CJames95** [comment]
Yeah, I should be able to recreate an example of what I was dealing with in code sandbox. It might take a day or two unfortunately, but I'll post a link to it when I have it up for you to see.

**Comment by CJames95** [comment]
Sorry for the wait. I have the code sandbox environment mostly working. Code sandbox has some weird quirks I'm unfamiliar with with how it renders the DOM versus what I'm used to, but hopefully this indicates the difference between what I was trying to convey. 

[Code Sandbox link](https://codesandbox.io/p/sandbox/shave-example-4397s5?layout=%257B%2522sidebarPanel%2522%253A%2522EXPLORER%2522%252C%2522rootPanelGroup%2522%253A%257B%2522direction%2522%253A%2522horizontal%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522id%2522%253A%2522ROOT_LAYOUT%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522clx1wgtnn0006356ibsutkb6u%2522%252C%2522sizes%2522%253A%255B100%252C0%255D%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522EDITOR%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522id%2522%253A%2522clx1wgtnn0002356ijmcjya6x%2522%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522SHELLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522id%2522%253A%2522clx1wgtnn0003356ipzkd60l2%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522DEVTOOLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522id%2522%253A%2522clx1wgtnn0005356ipalj4bwu%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%252C%2522sizes%2522%253A%255B50%252C50%255D%257D%252C%2522tabbedPanels%2522%253A%257B%2522clx1wgtnn0002356ijmcjya6x%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clx1wgtnn0001356ig28x8tby%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252Fsrc%252Findex.tsx%2522%257D%255D%252C%2522id%2522%253A%2522clx1wgtnn0002356ijmcjya6x%2522%252C%2522activeTabId%2522%253A%2522clx1wgtnn0001356ig28x8tby%2522%257D%252C%2522clx1wgtnn0005356ipalj4bwu%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clx1wgtnn0004356ifqogms5t%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522UNASSIGNED_PORT%2522%252C%2522port%2522%253A0%252C%2522path%2522%253A%2522%252F%2522%257D%255D%252C%2522id%2522%253A%2522clx1wgtnn0005356ipalj4bwu%2522%252C%2522activeTabId%2522%253A%2522clx1wgtnn0004356ifqogms5t%2522%257D%252C%2522clx1wgtnn0003356ipzkd60l2%2522%253A%257B%2522tabs%2522%253A%255B%255D%252C%2522id%2522%253A%2522clx1wgtnn0003356ipzkd60l2%2522%257D%257D%252C%2522showDevtools%2522%253Atrue%252C%2522showShells%2522%253Afalse%252C%2522showSidebar%2522%253Atrue%252C%2522sidebarPanelSize%2522%253A15%257D)
